code/modules/hydroponics/hydroponics_mutations.dm

```diff
@@ -0,0 +1,529 @@
+
+		if("change_appearance")
+			seed.randomize_icon(change_packet=0)
+			update_icon()
+			visible_message("<span class='notice'>\The [seed.display_name] suddenly looks a little different.</span>")
+
+		if("fruit_exotic")
+			seed.products += pick(typesof(/obj/item/weapon/reagent_containers/food/snacks/grown)-/obj/item/weapon/reagent_containers/food/snacks/grown)
+			visible_message("<span class='notice'>\The [seed.display_name] seems to be growing something weird.</span>")
+
+		if("chemical_exotic")
+			seed.add_random_chemical(severity)
+			visible_message("<span class='notice'>\The [seed.display_name] develops a strange-looking gland...</span>")
+
+		if("trait_hematophage")
+			seed.hematophage = !seed.hematophage
+			if(seed.hematophage)
+				visible_message("<span class='notice'>\The [seed.display_name] shudders thirstily, turning red at the roots!</span>")
+				nutrilevel = 1
+			else
+				visible_message("<span class='notice'>\The [seed.display_name]'s red roots slowly wash their color out...</span>")
+
+		if("trait_teleporting")
+			seed.teleporting = !seed.teleporting
+			if(seed.teleporting)
+				visible_message("<span class='notice'>\The [seed.display_name] wobbles unstably, glowing blue for a moment!</span>")
+			else
+				visible_message("<span class='notice'>\The [seed.display_name] slowly becomes spatial-temporally stable again.</span>")
+
+		if("trait_ligneous")
+			seed.ligneous = !seed.ligneous
+			if(seed.hematophage)
+				visible_message("<span class='notice'>\The [seed.display_name] seems to grow a cover of robust bark.</span>")
+			else
+				visible_message("<span class='notice'>\The [seed.display_name]'s bark slowly sheds away...</span>")
+
+		if("trait_juicy")
+			seed.juicy = seed.juicy ? 0 : 1
+			show_mutation_message()
+
+		if("trait_slippery")
+			switch(seed.juicy)
+				if(0 to 1)
+					seed.juicy = 2
+				if(2)
+					seed.juicy = 0
+			show_mutation_message()
+
+		if("trait_thorny")
+			seed.thorny = !seed.thorny
+			if(seed.thorny)
+				visible_message("<span class='notice'>\The [seed.display_name] spontaneously develops mean-looking thorns!</span>")
+			else
+				visible_message("<span class='notice'>\The [seed.display_name] sheds it's thorns away...</span>")
+
+		if("trait_stinging")
+			seed.stinging = !seed.stinging
+			if(seed.stinging)
+				visible_message("<span class='notice'>\The [seed.display_name] sprouts a coat of chemical stingers!</span>")
+			else
+				visible_message("<span class='notice'>\The [seed.display_name]'s stingers dry off and break...</span>")
+
+		else
+			visible_message("Looks like Bauds fucked up! Go tell him: [mutation_type]")
+
+	return
+	//visible_message("<span class='notice'>\The [seed.display_name] quivers!</span>")
+
+	/*//This looks like shit, but it's a lot easier to read/change this way.
+	var/total_mutations = rand(1,1+degree)
+	for(var/i = 0;i<total_mutations;i++)
```

Comment:
Ah damnit syntax highlighting failed me.

(If you look at the highlighting, it's counting the ' as the END of the comment?)